src/olympia/devhub/file_validation_annotations.py

```diff
@@ -30,135 +27,29 @@ def insert_validation_message(results, type_='error', message='', msg_id='',
-        # version_int('') is actually 200100. If strict compatibility is true,
-        # the validator should have complained about the non-existant max
-        # version, but it doesn't hurt to check that the value is sane anyway.
-        max_target_firefox_version > 200100 and
-        max_target_firefox_version < 53000000000000
-    )
-    is_targeting_firefox_higher_or_equal_than_57 = (
-        max_target_firefox_version >= 57000000000000 and
-        max_target_firefox_version < 99000000000000)
-
-    # Thunderbird/Seamonkey only add-ons are moving to addons.thunderbird.net.
-    if (is_targeting_thunderbird_or_seamonkey_only):
-        msg = ugettext(
-            u'Add-ons for Thunderbird and SeaMonkey are now listed and '
-            u'maintained on addons.thunderbird.net. You can use the same '
-            u'account to update your add-ons on the new site.')
-
-        insert_validation_message(
-            results, message=msg, msg_id='thunderbird_and_seamonkey_migration')
-
-    # Legacy add-ons are no longer supported on AMO (if waffle is enabled).
-    elif (is_extension_or_complete_theme and
-            waffle.switch_is_active('disallow-legacy-submissions')):
-        msg = ugettext(
-            u'Legacy extensions are no longer supported in Firefox.')
-
-        insert_validation_message(
-            results, message=msg, msg_id='legacy_addons_unsupported')
-
-    # New legacy add-ons targeting Firefox only must target Firefox 53 or
-    # lower, strictly. Extensions targeting multiple other apps are exempt from
-    # this.
-    elif (is_new_upload and
-            is_extension_or_complete_theme and
-            is_targeting_firefoxes_only and
-            not is_targeting_firefox_lower_than_53_only):
-
-        msg = ugettext(
-            u'Starting with Firefox 53, new add-ons on this site can '
-            u'only be WebExtensions.')
-
-        insert_validation_message(
-            results, message=msg, msg_id='legacy_addons_restricted')
-
-    # All legacy add-ons (new or upgrades) targeting Firefox must target
-    # Firefox 56.* or lower, even if they target multiple apps.
-    elif (is_extension_or_complete_theme and
-            is_targeting_firefox_higher_or_equal_than_57):
-        # Note: legacy add-ons targeting '*' (which is the default for sdk
-        # add-ons) are excluded from this error, and instead are silently
-        # rewritten as supporting '56.*' in the manifest parsing code.
-        msg = ugettext(
-            u'Legacy add-ons are not compatible with Firefox 57 or higher. '
-            u'Use a maxVersion of 56.* or lower.')
-
-        insert_validation_message(
-            results, message=msg, msg_id='legacy_addons_max_version')
+    description = ugettext(
+        u'Add-ons for Thunderbird and SeaMonkey are now listed and '
+        u'maintained on addons.thunderbird.net. You can use the same '
+        u'account to update your add-ons on the new site.')
```

Comment:
Not sure I understand here - the message is that legacy add-ons aren't supported but the description is about Thunderbird and Seamonkey add-ons.  But I don't see any code (remaining) that checks if the add-on is acutally targeting Thunderbird/Seamonkey.

Reply:
Given that Thunderbird/Seamonkey don't support WebExtensions at all I thought we don't necessarily have to double-check them all.

Maybe it's worth having a test just for sake of completeness that tests thunderbird/seamonkey?

Reply:
Probably.  It's just the way I read it was a developer uploads a (Firefox compatible) legacy add-on and we tell them about Thunderbird and Seamonkey, which would be confusing

Reply:
Indeed, yeah, I merged the messages because in the end it's all legacy add-ons and we don't support them. I'd even guess that most add-ons that run into that barrier may even be Thunderbird add-ons given that Firefox doesn't have support for legacy add-ons for quite some time.

Reply:
I forgot to come back to this before I r+'d - I don't think we should be returning an error message mentioning Thunderbird or Seamonkey unless the developer has specified Thunderbird or Seamonkey in the install.rdf.  Can we leave that check in?  (iirc the app compatibility is in addons-server rather than the validator)